src/services/DeliverablesService.ts

```diff
@@ -84,10 +89,31 @@ const list = async (
   );
 };
 
+/**
+ * Uploads multiple documents and waits for all promises to settle.
+ */
+const upload = async (deliverableId: number, documents: UploadDeliverableDocumentRequest[]): Promise<boolean> => {
+  const headers = { 'content-type': 'multipart/form-data' };
+  const promises = documents.map((document) => {
+    const { description, file, projectId } = document;
+    const urlReplacements = {
+      '{deliverableId}': `${deliverableId}`,
+      '{projectId}': `${projectId}`,
+    };
+    const entity = { description, file };
+
+    return httpDocumentUpload.post({ urlReplacements, entity, headers });
+  });
+
+  const results = await getPromisesResponse<Response>(promises);
+  return results.every((result) => result !== null && result.requestSucceeded === true);
+};
+
```

Comment:
How do we handle a partial failure? (one or more document uploads completed successfully, while one or more documents uploads failed)

Reply:
We don't have this pattern today, something we can add later. For now if any one fails, we show an error but we refresh the documents list so user knows which ones succeeded.